components/Buttons/tests/unit/MDCButtonTests.m

```diff
@@ -590,6 +590,77 @@ - (void)testTitleFontForStateWithLegacyDynamicTypeRestoresUserFonts {
                 self.button.titleLabel.font);
 }
 
+- (void)testTitleFontForStateDisabledAfterSettingFontsPreventsFontChange {
+  // Given
+  UIFont *normalFont = [UIFont systemFontOfSize:10];
+  UIFont *selectedFont = [UIFont systemFontOfSize:40];
+  UIFont *directlyAssignedFont = [UIFont systemFontOfSize:25];
+  [self.button setTitleFont:normalFont forState:UIControlStateNormal];
+  [self.button setTitleFont:selectedFont forState:UIControlStateSelected];
+
+  // When
+  self.button.enableTitleFontForState = NO;
+  self.button.titleLabel.font = directlyAssignedFont;
+  self.button.selected = YES;
+
+  // Then
+  XCTAssertEqualObjects(self.button.titleLabel.font, directlyAssignedFont);
```

Comment:
I think a comment needs to be added to this test to explain the reason we dont want the font to change.
A vanilla read of the test just seems to be confirming incorrect behavior: that the font does not change to the selectedFont specified by setTitleFont:forState:Selected.

In person you mentioned that its for dynamic type reasons but we see no dynamic type explanation here.

Reply:
We don't want the font to change because the API to enable that behavior was set to `NO`: `self.button.enableTitleFontForState = NO;`.   Can you help me understand what is hard to read?

## When
1. `titleFontForState` is disabled
2. and `titleLabel.font` is assigned
3. and the button's state changes

## Then
The button's title label's font does not get changed.

Reply:
oh right. i got confused I forgot that your testing the enableTitleFontForSate = NO

Perhaps a comment:
// This tests checks enableTitleFontForState = NO

Reply:
Done. Added a comment.